### ISSUE TITLE
"vue/valid-v-slot" rule support config custom element name to override some element name in html-elements.json

### ISSUE THREAD

**Issue by 643104191** [vuejs/eslint-plugin-vue#2113]
**What rule do you want to change?**
vue/valid-v-slot

**Does this change cause the rule to produce more or fewer warnings?**
yes, it can cause less warnings

**How will the change be implemented? (New option, new default behavior, etc.)?**
New option
```jsonc
{
  "vue/valid-v-slot": ["error", {
    // ...other options
    "customElement": ["map"]
  }]
}
```

**Please provide some example code that this change will affect:**
in uni-app vue3 project,there is a global Built-in components named ["map"](https://en.uniapp.dcloud.io/component/map.html)
it has many slots, such as `callout`
the `vue/valid-v-slot` rule will report an error(or warning) when i write template code like this:

```vue
<template>
  <map>
    <template #callout> <!-- 'v-slot' directive must be owned by a custom element, but 'map' is not.eslint(vue/valid-v-slot) -->
      <view>the code has been omitted</view>
    </template>
  </map>
</template>
```

**What does the rule currently do for this code?**
I’m trying to figure out why it can cause an error or a warning.

there is a `isCustomComponent` check in [`/lib/utils/index.js`](https://github.com/vuejs/eslint-plugin-vue/blob/master/lib/utils/index.js).

The `map` element name is not recognized as a custom element here, and the "vue/valid-v-slot" rule prohibits html element from using v-slot.

**What will the rule do after it's changed?**
don't report an error(or warning) when i write template code like this:
```vue
<template>
  <map>
    <template #callout>
      <view>the code has been omitted</view>
    </template>
  </map>
</template>
```

**Additional context**

**Comment by ota-meshi** [comment]
This plugin assumes that elements defined in HTML are not components.
https://developer.mozilla.org/ja/docs/Web/HTML/Element/map
It is recommended not to use a component with the same name as an HTML element.

**Comment by 643104191** [comment]
> This plugin assumes that elements defined in HTML are not components. https://developer.mozilla.org/ja/docs/Web/HTML/Element/map It is recommended not to use a component with the same name as an HTML element.

@ota-meshi 
I don’t want to use a component that has the same name as an HTML element.
This is because uni-app has built-in components named “map”.
Therefore, the rule config is very important for uni-app developers.
